internal/k8s/crdcontroller.go

```diff
@@ -208,3 +219,214 @@ func (c *AviController) SetupAKOCRDEventHandlers(numWorkers uint32) {
 
 	return
 }
+
+// validateHostRuleObj would do validation checks
+// update internal CRD caches, and push relevant ingresses to ingestion
+func ValidateHostRuleObj(key string, hostrule *akov1alpha1.HostRule) error {
```

Comment:
I don't see the return value getting used anymore. We can skip it.

Reply:
Actually, it would probably be better to return the error from here and print it in the caller function IMHO. What do you think @sudswasavi @monotosh-avi